app/src/main/java/fr/neamar/kiss/UiTweaks.java

```diff
@@ -0,0 +1,71 @@
+package fr.neamar.kiss;
+
+import android.app.ActionBar;
+import android.app.Activity;
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.view.Window;
+import android.view.WindowManager;
+import android.widget.ImageView;
+
+public class UiTweaks {
+    static void updateThemePrimaryColor(Activity activity) {
+        String primaryColorOverride = getPrimaryColor(activity);
+
+        // Circuit breaker, keep default behavior.
+        if (primaryColorOverride.equals("#4caf50")) {
+            return;
+        }
+
+        int primaryColor = Color.parseColor(primaryColorOverride);
+
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            Window window = activity.getWindow();
+            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+
+            // Update status bar color
+            window.setStatusBarColor(primaryColor);
+        }
+
+        ActionBar actionBar = activity.getActionBar();
+        if (actionBar != null) {
+            actionBar.setBackgroundDrawable(new ColorDrawable(primaryColor));
+        }
+    }
+
+    static void tintResources(MainActivity mainActivity) {
+        String primaryColorOverride = getPrimaryColorForDisplay(mainActivity);
+
+        // Circuit breaker, keep default behavior.
+        if (primaryColorOverride.equals("#4caf50")) {
```

Comment:
I tried to use the constant value @color/kiss_green, but then Settings crash because they expect a string reference.

I did not find a way to read the defaultValue from code, ut if you know it I'm interested!